mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,442 @@
+        plt.show()
+
+
+class HetHomRatioPerSample(object):
+    """
+    Plot Histogram of Het/Hom ratio per sample
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext 
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data_het =  df.where( (df.alleles[0] == u'REF') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        hets =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_het }
+        data_hom =  df.where( (df.alleles[0] == u'ALT') & (df.alleles[1] == u'ALT')).groupBy("sampleid").count().collect()
+        homs =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_hom }
+        self.het_hom_ratio = [ float(hets[sampleid])/float(homs[sampleid]) for sampleid in homs.keys() ]
+
+    def plot(self, testMode = False):
+        if (not testMode):
+          plt.hist(self.het_hom_ratio,bins=100)
+          plt.title("Histogram of Het/Hom Ratio")
+          plt.xlabel("Het/Hom Ratio")
+          plt.ylabel("Sample Count")
+          plt.show()
+        return self.het_hom_ratio
+
+
+class CallRatePerSample(object):
+    """
+    Plot Histogram of call rate per sample
+    """
+    def __init__(self,ss,genotypeRDD):
+        """
+        :param ss: SparkContext 
+        :param genotypeRDD: bdgenomics genotypeRDD
+        """
+        df =  genotypeRDD.toDF()
+        data_called =  df.where( (df.alleles[0] != u'NO_CALL')).groupBy("sampleid").count().collect()
+        called =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_called }
+        data_missing =  df.where( (df.alleles[0] == u'NO_CALL')).groupBy("sampleid").count().collect()
+        missing =  { sample.asDict()['sampleid']:sample.asDict()['count'] for sample in data_missing }
+        self.call_rate = [ float(called[sampleid])/(float(missing[sampleid]+float(called[sampleid]))) for sampleid in called.keys() ]
+
+    def plot(self, testMode = False):
+        if (not testMode):
+          plt.hist(self.call_rate,bins=100)
+          plt.title("Histogram of Call Rate per sample")
+          plt.xlabel("Call Rate")
+          plt.ylabel("Sample Count")
+          plt.show()
+        return self.call_rate
+
+
+
+class QDDist(object):
+    """
+    Plot Histogram of Quality By Depth (QD) score from a variant dataset
+    """
+    def __init__(self,ss,variantRDD):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics variantRDD
+        """
+        data = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).map(lambda x: math.floor(float(x['attributes']['QD'])*10/10)).countByValue()
+        self.scores = [ x[0] for x in data.items() ]
+        self.weights = [ x[1] for x in data.items() ]
+
+    def plot(self, testMode = False):
+        if (not testMode):
+          plt.hist(self.scores,weights=self.weights, bins=80)
```

Comment:
why bins=80?

Reply:
It seemed to make a readable histogram for a typical distribution of this QD value

Reply:
should it be an option?